docs/mkdocs/guides/contributing/setup.md

```diff
@@ -0,0 +1,208 @@
+
+```bash
+source env/bin/activate
+```
+
+To deactivate the virtual environment, you can type:
+
+```bash
+deactivate
+```
+
+!!! warning
+
+    If you use a different name for this virtual environment subfolder, you will need to register the new name in
+    several config files including `.gitignore`, `.flake8`, `pyproject.toml` lest it will create friction for
+    other workflows such as pre-commit. Hence, we strongly recommend you name the subfolder `env/` as instructed above.
+
+### Installing Dependencies
+
+Once the virtual environment is activated, install packages necessary for running and developing `lineapy`:
+
+```bash
+pip install --upgrade pip
+pip install --upgrade setuptools
+pip install -r requirements.txt
+pip install -r docs/requirements.txt
+pip install -e '.[dev]'
+```
+
+Note that this may take a while to complete.
+
+## Running LineaPy
+
+During development, you may want to check how your changes affect the behavior of the package.
+LineaPy supports several interfaces to run on, and the relevant instructions are documented [here](../setup.md#running-lineapy).
+Please select the most convenient option for you.
+
+!!! note
+
+    When running LineaPy, make sure that the virtual environment has been activated.
+
+## Development
+
+### Making Changes
+
+With development dependencies installed, you are now ready to contribute to the source code!
+First, make a separate branch for your development work. Please use an informative name so that
+others can get good sense of what your changes are about.
+
+```bash
+git checkout -b <NEW-BRANCH-NAME>
+```
+
+After making changes you desire, save them to your development branch:
+
+```bash
+git add <PATH-TO-CHANGED-FILE>
+git commit -m "<COMMIT-MESSAGE>"
+```
+
+!!! info
+
+    To learn more about saving changes in Git, check this [tutorial](https://www.atlassian.com/git/tutorials/saving-changes).
+
+!!! tip
+
+    LineaPy provides several [pre-commit](https://pre-commit.com/) hooks to automatically standardize styles and formats
+    across its codebase.
+
+    To run these hooks automatically upon every new commit:
+
+    ```bash
+    pre-commit install
+    ```
+
+    To run the hooks even when there are no changes:
+
+    ```bash
+    pre-commit run --all-files
+    ```
+
+    LineaPy's pre-commit hooks do not run tests since tests are too time-consuming to run on every commit. For testing, see the section below.
+
+Note that these changes have been saved only locally at this point, and you need to "push" them to your forked repo on GitHub:
+
+```bash
+git push
+```
+
+If the new (development) branch has not been pushed before, you will need to create its counterpart on GitHub with:
+
+```bash
+git push --set-upstream origin <NEW-BRANCH-NAME>
+```
+
+### Documenting Changes
+
+Good code documentation is essential to effective collaboration among different developers. As such,
+we ask contributors to add proper [NumPy-styled](https://numpydoc.readthedocs.io/en/latest/format.html)
```

Comment:
@aayan636 can you provide an example? I prefer to skip this since we are not enforcing this very strictly now, and we can come back to it if we enforce strictly our code documentation style.